File: ios/CreatePrivateDealerOS/CreatePrivateDealerOS/Models/DealerRole.swift
import Foundation

enum DealerRole: String, Codable, CaseIterable {
    case dealerAdmin = "dealer_admin"
    case dealerSales = "dealer_sales"
    case dealerMarketing = "dealer_marketing"
    case platformOwner = "platform_owner"
    case platformSupport = "platform_support"

    var canManageSettings: Bool {
        self == .dealerAdmin || self == .platformOwner || self == .platformSupport
    }

    var canWriteConversations: Bool {
        self == .dealerAdmin || self == .dealerSales || self == .platformOwner || self == .platformSupport
    }

    func canViewMessageContent(marketingFeatureFlag: Bool) -> Bool {
        switch self {
        case .dealerAdmin, .dealerSales, .platformOwner, .platformSupport:
            return true
        case .dealerMarketing:
            return marketingFeatureFlag
        }
    }

    var tabItems: [AppTab] {
        switch self {
        case .dealerAdmin:
            return [.dashboard, .conversations, .leads, .bookings, .analytics, .settings]
        case .dealerSales:
            return [.dashboard, .conversations, .leads, .bookings, .analytics]
        case .dealerMarketing:
            return [.dashboard, .conversations, .leads, .bookings, .analytics]
        case .platformOwner, .platformSupport:
            return [.dashboard, .analytics]
        }
    }
}

File: flutter/create_private_dealer_os/lib/domain/models/dealer_role.dart
import '../../core/routing/app_tab.dart';

enum DealerRole {
  dealerAdmin('dealer_admin'),
  dealerSales('dealer_sales'),
  dealerMarketing('dealer_marketing'),
  platformOwner('platform_owner'),
  platformSupport('platform_support');

  const DealerRole(this.value);
  final String value;

  static DealerRole fromString(String raw) {
    return DealerRole.values.firstWhere(
      (role) => role.value == raw,
      orElse: () => DealerRole.dealerSales,
    );
  }

  bool get canManageSettings {
    return this == DealerRole.dealerAdmin ||
        this == DealerRole.platformOwner ||
        this == DealerRole.platformSupport;
  }

  bool get canWriteConversations {
    return this == DealerRole.dealerAdmin ||
        this == DealerRole.dealerSales ||
        this == DealerRole.platformOwner ||
        this == DealerRole.platformSupport;
  }

  bool canViewMessageContent({required bool marketingFeatureFlag}) {
    switch (this) {
      case DealerRole.dealerAdmin:
      case DealerRole.dealerSales:
      case DealerRole.platformOwner:
      case DealerRole.platformSupport:
        return true;
      case DealerRole.dealerMarketing:
        return marketingFeatureFlag;
    }
  }

  List<AppTab> get tabs {
    switch (this) {
      case DealerRole.dealerAdmin:
        return <AppTab>[
          AppTab.dashboard,
          AppTab.conversations,
          AppTab.leads,
          AppTab.bookings,
          AppTab.analytics,
          AppTab.settings,
        ];
      case DealerRole.dealerSales:
      case DealerRole.dealerMarketing:
        return <AppTab>[
          AppTab.dashboard,
          AppTab.conversations,
          AppTab.leads,
          AppTab.bookings,
          AppTab.analytics,
        ];
      case DealerRole.platformOwner:
      case DealerRole.platformSupport:
        return <AppTab>[AppTab.dashboard, AppTab.analytics];
    }
  }
}
